modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ExecutionServiceImpl.java

```diff
@@ -667,12 +691,15 @@ private FieldsMetadata explainFieldsMetadata(PlanningContext ctx) {
             plan.remotes(fragment));
 
         ExecutionContext<Row> ectx = new ExecutionContext<>(
+            qctx,
             taskExecutor(),
-            pctx,
             qryId,
+            locNodeId,
+            locNodeId,
+            topologyVersion(),
```

Comment:
Topology version can be changed between `plan.init()` and `ectx` creation, in this case, query will be mapped to the wrong nodes and not fail on the execution step. Let's store `topologyVersion()` to variable and use it in both calls.

Reply:
Thanks for attention! Fixed.